### ISSUE TITLE
Request not made using Oauth 2 Client Credentials

### ISSUE THREAD

**Issue by PabloLec** [usebruno/bruno#2074]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

I am using the GUI on latest version 1.13.0.

I am having an issue with OAuth2 using Client Credentials grant type:
When I initiate the request, I only get the result from the request to the access token URL, not the result from the main request.
The JSON with the access_token is displayed but nothing else; the main request either isn't being made, or at least its result isn't displayed in the response or in the timeline.

Moreover, the wrong URL is displayed in the timeline, I see my main request URL not my IAM/Oauth 2 server URL. Despite the call being made to the right IAM/Oauth 2 URL as I can see a successful response with access_token in the "Response" window.

### Screenshots/Live demo link

As an example here is the timeline, with a call to my IAM for the access_token and nothing more:

```
200 OK
49ms
1.31KB
> GET **MAIN REQUEST URL / NOT IAM OAUTH2 UR**
> data {"grant_type":"client_credentials","client_id":"***","client_secret":"***","scope":"***"}
< 200 OK
< cache-control: no-store
< content-length: 1342
< content-type: application/json
< pragma: no-cache
< referrer-policy: no-referrer
< x-content-type-options: nosniff
< x-frame-options: SAMEORIGIN
< x-xss-protection: 1; mode=block
< date: Sat, 13 Apr 2024 17:10:55 GMT
< strict-transport-security: max-age=31536000; includeSubDomains; preload
< access-control-allow-origin: none
```

**Comment by pietrygamat** [comment]
This is duplicate of #1999 and I am currently experimenting with the fix in #2061. I would appreciate you tried running your request on a version build from that code. That's unless @helloanoop is not baking something entirely different that is ;).

**Comment by PabloLec** [comment]
@pie

Thanks, will close this one in favor of #1999
